jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/AbstractAvailableJobsExecutor.java

```diff
@@ -0,0 +1,255 @@
+                CommandContext ctx = null;
+                List<CommandCallback> callbacks = null;
+                ClassLoader cl = getClassLoader(request.getDeploymentId());
+                try {
+    
+                    logger.debug("Processing Request Id: {}, status {} command {}", request.getId(), request.getStatus(), request.getCommandName());
+                    
+                    
+                    byte[] reqData = request.getRequestData();
+                    if (reqData != null) {
+                        ObjectInputStream in = null;
+                        try {
+                            in = new ClassLoaderObjectInputStream(cl, new ByteArrayInputStream(reqData));
+                            ctx = (CommandContext) in.readObject();
+                        } catch (IOException e) {                        
+                            logger.warn("Exception while serializing context data", e);
+                            return;
+                        } finally {
+                            if (in != null) {
+                                in.close();
+                            }
+                        }
+                    }
+                    for (Map.Entry<String, Object> entry : contextData.entrySet()) {
+                    	ctx.setData(entry.getKey(), entry.getValue());
+                    }
+                    // add class loader so internally classes can be created with valid (kjar) deployment
+                    ctx.setData("ClassLoader", cl);
+                    
+                    
+                    cmd = classCacheManager.findCommand(request.getCommandName(), cl);
+                    ExecutionResults results = cmd.execute(ctx);
+                    
+                    callbacks = classCacheManager.buildCommandCallback(ctx, cl);                
+                    
+                    for (CommandCallback handler : callbacks) {
+                        
+                        handler.onCommandDone(ctx, results);
+                    }
+                    
+                    if (results != null) {
+                        try {
+                            ByteArrayOutputStream bout = new ByteArrayOutputStream();
+                            ObjectOutputStream out = new ObjectOutputStream(bout);
+                            out.writeObject(results);
+                            byte[] respData = bout.toByteArray();
+                            request.setResponseData(respData);
+                        } catch (IOException e) {
+                            request.setResponseData(null);
+                        }
+                    }
+    
+                    request.setStatus(STATUS.DONE);
+                     
+                    executorStoreService.updateRequest(request);
+                    processReoccurring = true;
+                    
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
```

Comment:
it can be thrown by bitronix on shutdown of the transaction manager, not sure if that can happen on regular app servers but if the thread was not interrupted it affected all other test executions